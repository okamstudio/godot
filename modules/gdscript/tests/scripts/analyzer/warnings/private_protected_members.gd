class A:
    @warning_ignore("unused_protected_class_variable")
    static var _static_a = null
    @warning_ignore("unused_private_class_variable")
    static var __static_b = null

    @warning_ignore("unused_protected_class_variable")
    var _a = null
    @warning_ignore("unused_private_class_variable")
    var __b = null

    func _call_a():
        pass

    func __call_b():
        pass

    func _call_a_ret():
        return null

    func __call_b_ret():
        return null

    static func _static_call_a():
        pass

    static func __static_call_b():
        pass

    static func _static_call_a_ret():
        return null

    static func __static_call_b_ret():
        return null

class B:
    func _init():
        var cls_a = A.new()
        A._static_a = null
        A.__static_b = null
        cls_a._a = null
        cls_a.__b = null
        cls_a._call_a()
        cls_a.__call_b()

        @warning_ignore("unused_variable")
        var t1 = cls_a._call_a_ret()
        @warning_ignore("unused_variable")
        var t2 = cls_a.__call_b_ret()

        A._static_call_a()
        A.__static_call_b()
        @warning_ignore("unused_variable")
        var t3 = A._static_call_a_ret()
        @warning_ignore("unused_variable")
        var t4 = A.__static_call_b_ret()

class C extends A:
    func _init():
        A._static_a = null
        A.__static_b = null
        _a = null
        __b = null
        _call_a()
        __call_b()

        @warning_ignore("unused_variable")
        var t1 = _call_a_ret()
        @warning_ignore("unused_variable")
        var t2 = __call_b_ret()

        _static_call_a()
        __static_call_b()
        @warning_ignore("unused_variable")
        var t3 = _static_call_a_ret()
        @warning_ignore("unused_variable")
        var t4 = __static_call_b_ret()

func test():
    pass
