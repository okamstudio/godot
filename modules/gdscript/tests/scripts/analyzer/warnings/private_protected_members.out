GDTEST_OK
~~ WARNING at line 39: (ACCESSING_PROTECTED_MEMBER) Trying accessing the member "_static_a" prefixed with "_", which would be a protected member.
~~ WARNING at line 40: (ACCESSING_PRIVATE_MEMBER) Trying accessing the member "__static_b" prefixed with "__", which would be a private member.
~~ WARNING at line 41: (ACCESSING_PROTECTED_MEMBER) Trying accessing the member "_a" prefixed with "_", which would be a protected member.
~~ WARNING at line 42: (ACCESSING_PRIVATE_MEMBER) Trying accessing the member "__b" prefixed with "__", which would be a private member.
~~ WARNING at line 43: (CALLING_PROTECTED_METHOD) Trying calling the method "_call_a()" prefixed with "_", which would be a protected method.
~~ WARNING at line 44: (CALLING_PRIVATE_METHOD) Trying calling the method "__call_b()" prefixed with "__", which would be a private method.
~~ WARNING at line 47: (CALLING_PROTECTED_METHOD) Trying calling the method "_call_a_ret()" prefixed with "_", which would be a protected method.
~~ WARNING at line 49: (CALLING_PRIVATE_METHOD) Trying calling the method "__call_b_ret()" prefixed with "__", which would be a private method.
~~ WARNING at line 51: (CALLING_PROTECTED_METHOD) Trying calling the method "_static_call_a()" prefixed with "_", which would be a protected method.
~~ WARNING at line 52: (CALLING_PRIVATE_METHOD) Trying calling the method "__static_call_b()" prefixed with "__", which would be a private method.
~~ WARNING at line 54: (CALLING_PROTECTED_METHOD) Trying calling the method "_static_call_a_ret()" prefixed with "_", which would be a protected method.
~~ WARNING at line 56: (CALLING_PRIVATE_METHOD) Trying calling the method "__static_call_b_ret()" prefixed with "__", which would be a private method.
~~ WARNING at line 61: (ACCESSING_PRIVATE_MEMBER) Trying accessing the member "__static_b" prefixed with "__", which would be a private member.
~~ WARNING at line 63: (ACCESSING_PRIVATE_MEMBER) Trying accessing the member "__b" prefixed with "__", which would be a private member.
~~ WARNING at line 65: (CALLING_PRIVATE_METHOD) Trying calling the method "__call_b()" prefixed with "__", which would be a private method.
~~ WARNING at line 70: (CALLING_PRIVATE_METHOD) Trying calling the method "__call_b_ret()" prefixed with "__", which would be a private method.
~~ WARNING at line 73: (CALLING_PRIVATE_METHOD) Trying calling the method "__static_call_b()" prefixed with "__", which would be a private method.
~~ WARNING at line 77: (CALLING_PRIVATE_METHOD) Trying calling the method "__static_call_b_ret()" prefixed with "__", which would be a private method.
